libs/community/langchain_community/chains/azure_content_safety_chain.py

```diff
@@ -0,0 +1,163 @@
+
+            moderated_chain = moderate | prompt | model
+
+            moderated_chain.invoke({"input": "Hey, How are you?"})
+    """
+
+    client: Any = None  #: :meta private:
+    error: bool = False
+    """Whether or not to error if bad content was found."""
+    input_key: str = "input"  #: :meta private:
+    output_key: str = "output"  #: :meta private:
+    content_safety_key: Optional[str] = None
+    content_safety_endpoint: Optional[str] = None
+
+    @property
+    def input_keys(self) -> List[str]:
+        """Expect input key.
+
+        :meta private:
+        """
+        return [self.input_key]
+
+    @property
+    def output_keys(self) -> List[str]:
+        """Return output key.
+
+        :meta private:
+        """
+        return [self.output_key]
+
+    @model_validator(mode="before")
+    @classmethod
+    def validate_environment(cls, values: Dict) -> Any:
+        """Validate that api key and python package exists in environment."""
+        content_safety_key = get_from_dict_or_env(
+            values, "content_safety_key", "CONTENT_SAFETY_API_KEY"
+        )
+        content_safety_endpoint = get_from_dict_or_env(
+            values, "content_safety_endpoint", "CONTENT_SAFETY_ENDPOINT"
+        )
+        try:
+            import azure.ai.contentsafety as sdk
+            from azure.core.credentials import AzureKeyCredential
+
+            values["client"] = sdk.ContentSafetyClient(
+                endpoint=content_safety_endpoint,
+                credential=AzureKeyCredential(content_safety_key),
+            )
+
+        except ImportError:
+            raise ImportError(
+                "azure-ai-contentsafety is not installed. "
+                "Run `pip install azure-ai-contentsafety` to install."
+            )
+        return values
+
+    def _detect_harmful_content(self, text: str, results: Any) -> str:
+        contains_harmful_content = False
+
+        for category in results:
+            if category["severity"] > 0:
+                contains_harmful_content = True
+
+        if contains_harmful_content:
+            error_str = (
+                "The input text contains harmful content "
+                "according to Azure OpenAI's content policy"
+            )
+            if self.error:
+                raise AzureHarmfulContentError(input=text)
+            else:
+                return error_str
+
+        return text
+
+    def _call(
+        self,
+        inputs: Dict[str, Any],
+        run_manager: Optional[CallbackManagerForChainRun] = None,
+    ) -> Dict[str, Any]:
+        text = inputs[self.input_key]
+
+        from azure.ai.contentsafety.models import AnalyzeTextOptions
+
+        request = AnalyzeTextOptions(text=text)
+        response = self.client.analyze_text(request)
+
+        result = response.categories_analysis
+        output = self._detect_harmful_content(text, result)
+
+        return {self.input_key: output, self.output_key: output}
```

Comment:
What does returning the input_key: with the output do?

Reply:
This makes it so the input of the next step of the chain is the filtered content. It makes it so that if the user doesn't want an error to be thrown, the model will not receive harmful content but instead the message ''The input has breached Azure's content safety policy'. Without this the input at the next step would still be the original harmful content.